foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/util/BrowserUtility.java

```diff
@@ -66,4 +104,280 @@ public void openUrl(String url, IWebBrowser browser, String pluginId, ILog log)
 			log.log(errorStatus);
 		}
 	}
+	
+	static public void openExtenalBrowser(String URL) {
+		checkedCreateExternalBrowser(URL,FoundationUIPlugin.PLUGIN_ID,FoundationUIPlugin.getDefault().getLog());
+	}
+	
+	/**
+	 * Creates an SWT Browser
+	 * @param style
+	 * @param parent
+	 * @return
+	 */
+	
+	public static Browser createBrowser(int style, Composite parent) {
+		return createBrowser(style, parent, getPreferredBrowser());
+	}
+
+	/**
+	 * Creates an SWT Browser
+	 * @param parent
+	 * @param preferredBrowser SWT.MOZILLA, SWT.WEBKIT or SWT.NONE
+	 * @return
+	 */
+	public static Browser createBrowser(Composite parent) {
+		return createBrowser(SWT.READ_ONLY | SWT.NO_SCROLL, parent, getPreferredBrowser());
+	}
+	
+	/**
+	 * Creates an SWT Browser
+	 * @param parent
+	 * @param preferredBrowser SWT.MOZILLA, SWT.WEBKIT or SWT.NONE
+	 * @return
+	 */
+	public static Browser createBrowser(Composite parent, int preferredBrowser) {
+		return createBrowser(SWT.READ_ONLY | SWT.NO_SCROLL, parent, preferredBrowser);
+	}
+
+	/**
+	 * Creates an SWT Browser. Returns null if can't create a browser.
+	 * @param style
+	 * @param parent
+	 * @param preferredBrowser SWT.MOZILLA, SWT.WEBKIT or SWT.NONE
+	 * @return
+	 * @see org.eclipse.swt.browser.Browser
+	 */
+	public static Browser createBrowser(int style, Composite parent, int preferredBrowser) {
+		/*
+			// We can provide webkit in this way:
+			   String defaultBrowser = System.getProperty(Browser.PROPERTY_DEFAULTTYPE);
+			   boolean hasDefaultBrowser = defaultBrowser != null;
+			   System.getProperties().setProperty(Browser.PROPERTY_DEFAULTTYPE, "webkit");
+		*/
+		if(System.getProperty("skipBrowserCreation")!=null) {
```

Comment:
we should not let code into foundation without knowing why it is there.

if it is going to be there then follow the naming pattern/conventions we have elsewhere.

* lowercase
* "scoped" by package/feature

so that would be something like 'jbosstools.skip.browser.creation'  even though I can see it seems in foundation there is already several variations; but lets not make it worse.

and since it is a boolan it should not accept skipBrowserCreation=false